File: lab-one/src/LabOne.java
import java.util.Scanner;

public class LabOne {

    private static String SEPARATOR = "--------------------------------------------------";
    private static String WELCOME = "Выберите часть лабы (1/2). Для выхода введите 'q'.";
    private static String PART_ONE = "1";
    private static String PART_TWO = "2";
    private static String QUIT = "q";
    private static String ERROR = "Ошибочный ввод";

    public static void main(String[] args) {

        Scanner in = new Scanner(System.in);
        while (true) {
            System.out.println(SEPARATOR);
            System.out.println(WELCOME);
            String partNumber = in.nextLine();
            if (partNumber.equals(PART_ONE)) {
                (new PartOne()).run();
            } else if (partNumber.equals(PART_TWO)) {
                (new PartTwo()).run();
            } else if (partNumber.equals(QUIT)) {
                break;
            } else {
                System.out.println(ERROR);
            }
        }
    }
}

File: LabOne/LabOne.cs
using System;

namespace LabOne
{
    public class LabOne
    {
        private const string Separator = "--------------------------------------------------";
        private const string Welcome = "Выберите часть лабы (1/2). Для выхода введите 'q'.";
        private const string PartOne = "1";
        private const string PartTwo = "2";
        private const string Quit = "q";
        private const string Error = "Ошибочный ввод";

        public static void Main(string[] args)
        {
            while (true)
            {
                Console.WriteLine(Separator);
                Console.WriteLine(Welcome);
                var partNumber = Console.ReadLine();
                if (partNumber == PartOne)
                    new PartOne().Run();
                else if (partNumber == PartTwo)
                    new PartTwo().Run();
                else if (partNumber == Quit)
                    break;
                else
                    Console.WriteLine(Error);
            }
        }
    }
}